### ISSUE TITLE
Destroy / empty cart when cross browser is enabled 

### ISSUE THREAD

**Issue by umbertix** [lukepolo/laracart#115]
Executing any of the following commands do not clear my cart.
`$cart = LaraCart::get();`
`$cart->emptyCart();`
`$cart->destroyCart();`
`LaraCart::emptyCart();`
`LaraCart::destroyCart();`

If I delete the files in the sessions folder obv. I get a clean cart. Am I missing any instruction? Or doing something wrong?

**Comment by lukepolo** [comment]
Those should be working, I have not ran into an issue, although im using redis and not file storage. I can try that and see if theres an issue.

**Comment by umbertix** [comment]
Running it with redis does not work either for me. Might be some missconfiguration on my machine? The  sesion is persisted correctly because I can access the current cart without a problem.

**Comment by lukepolo** [comment]
That seems very odd, are you running homestead?

**Comment by umbertix** [comment]
A vagrant machine, using the Homestad.yaml file.

**Comment by umbertix** [comment]
If I manually run: redis-cli flushall
The cart would be cleaned correctly, with all the rest of session and cache... :(

**Comment by lukepolo** [comment]
Well, this is odd. I have no idea why it would not work, my empty / delete function does this : 


 unset($this->cart->items);

 $this->session->set($this->prefix . '.' . $this->cart->instance, $this->cart);

        if (config('laracart.cross_devices', false) && $this->authManager->check()) {
            $this->authManager->user()->cart_session_id = $this->session->getId();
            $this->authManager->user()->save();
        }


If you want , follow these steps and do a dd to make sure the cart is empty before we save. 

Also make sure you dont have anywhere in your code re-adding the items.

**Comment by lukepolo** [comment]
Do you have cross devices config turned on ?

**Comment by umbertix** [comment]
Yes.

**Comment by lukepolo** [comment]
Try disabling that for now. And try to see if emptying works.

**Comment by umbertix** [comment]
Yep. That fix it.

**Comment by lukepolo** [comment]
K ill look at a fix for that. marking this a bug , changing title

**Comment by lukepolo** [comment]
Can you post some sample code how you were able to repeat this? I am unable to get this to work.

**Comment by lukepolo** [comment]
Steps ive tried :

```
\LaraCart::add(2, 'shirt', 200, 15.99, [
    'size' => 'XL'
]);

\LaraCart::getItems()
    // Output
    array:1 [▼
      "3f9b1508e6e6e83991b7081a4ea838fe" => CartItem {#351 ▶}
   ]
```

```
    // On the next request I empty the cart
    \LaraCart::emptyCart()
```

```
    // On the next request I try to get the items
    \LaraCart::getItems()

    // Output
    []
```

**Comment by umbertix** [comment]
//routes.php
```
Route::get('cart/add/{diamond_id}', ['uses' => 'CartController@addDiamond', 'as' => 'public.cart.addItem']);
Route::get('cart', ['uses' => 'CartController@index', 'as' => 'public.cart.index']);
Route::get('cart/destroy', ['uses' => 'CartController@destroy', 'as' => 'public.cart.destroy']);
```

//laracart.php
`cross_devices' => true`

//CartController.php
```

    public function index()
    {
        $cart = LaraCart::get();
        $cart_items = $cart->getItems();

        return view('public.cart.show', compact('cart_items'));
    }

    public function destroy()
    {
        LaraCart::emptyCart();
        LaraCart::destroyCart();
    }

    public function addDiamond($diamond_id)
    {
        $diam = Diamond::findOrFail($diamond_id);
        LaraCart::add($diam);
    }
```

//public/cart/show.blade.php
```
<table class="table table-striped">
        <thead>
        <tr>
            <th>Id</th>
            <th>Price</th>
        </tr>
        </thead>
        <tbody>
        @foreach($cart_items as $item)
            <tr>
                <td>{{ $item->id }}</td>
                <td>{{ $item->price }}</td>
            </tr>
        @endforeach
        </tbody>
    </table>
    Sub total: {{ LaraCart::subTotal($format = true, $withDiscount = true) }}<br>
    Total discount: {{ LaraCart::totalDiscount($formatted = true) }}<br>
    Tax total: {{ LaraCart::taxTotal($formatted = true) }}<br>
    Total: {{ LaraCart::total($formatted = true, $withDiscount = true) }}<br>
```

**Comment by lukepolo** [comment]
fixed via commit fd7c36fe9626064a02a3a91063e73736902a9f4d
